src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -630,6 +630,18 @@ end
 @[simp] lemma is_cycle.not_of_nil {u : V} : ¬ (nil : G.walk u u).is_cycle :=
 λ h, h.ne_nil rfl
 
+/-! ### About paths -/
+
+instance [decidable_eq V] {u v : V} (p : G.walk u v) : decidable p.is_path :=
+by { rw is_path_def, apply_instance }
+
+lemma is_path.length_lt [fintype V] {u v : V} {p : G.walk u v} (hp : p.is_path) :
+  p.length < fintype.card V :=
+begin
+  rw [nat.lt_iff_add_one_le, ← length_support],
+  exact hp.support_nodup.length_le_card,
+end
```

Comment:
```suggestion
by { rw [nat.lt_iff_add_one_le, ← length_support], exact hp.support_nodup.length_le_card }
```

Reply:
I'll put this into one line, but note that this kind of suggestion is outside the purview of a code review. We have some style guidelines, but we let contributors make their own choices here.